src/library/Table/__tests__/Table.spec.js

```diff
@@ -26,4 +110,398 @@ describe('Table', () => {
+
+        app.update();
+        table = app.find(Table);
+        const sortedData = table.find(TableBase).props().data;
+        const headerCellHtml = stripReactTextTags(headerCell.html());
+
+        expect(sortedData).toMatchSnapshot();
+        expect(headerCellHtml).toMatchSnapshot();
+      });
+
+      it('toggles sort direction on second click', () => {
+        const headerCell = table.find(TableSortableHeaderCell).first();
+        const button = headerCell.find('button');
+
+        button.simulate('click');
+        button.simulate('click');
+
+        app.update();
+        table = app.find(Table);
+        const sortedData = table.find(TableBase).props().data;
+        const headerCellHtml = stripReactTextTags(headerCell.html());
+
+        expect(sortedData).toMatchSnapshot();
+        expect(headerCellHtml).toMatchSnapshot();
+      });
+
+      it('sorts ascending when sorted column changes', () => {
+        const headerCell = table.find(TableSortableHeaderCell).first();
+        const secondHeaderCell = table.find(TableSortableHeaderCell).at(1);
+        const button = headerCell.find('button');
+        const secondButton = secondHeaderCell.find('button');
+
+        button.simulate('click');
+        secondButton.simulate('click');
+
+        app.update();
+        table = app.find(Table);
+        const sortedData = table.find(TableBase).props().data;
+        const headerCellHtml = stripReactTextTags(headerCell.html());
+        const secondHeaderCellHtml = stripReactTextTags(
+          secondHeaderCell.html()
+        );
+
+        expect(sortedData).toMatchSnapshot();
+        expect(headerCellHtml).toMatchSnapshot('Idle');
+        expect(secondHeaderCellHtml).toMatchSnapshot('Active');
+      });
+
+      it('calls onSort callback when provided', () => {
+        const button = table
+          .find(TableSortableHeaderCell)
+          .first()
+          .find('button');
+
+        button.simulate('click');
+
+        expect(onSort).toHaveBeenCalledWith({ key: 'aa', descending: false });
+      });
+    });
+
+    it('maintains sort state when data changes', () => {
+      const props = {
+        columns: [
+          { content: 'AA', key: 'aa' },
+          { content: 'AB', key: 'ab' },
+          { content: 'AC', key: 'ac' },
+          { content: 'AD', key: 'ad' }
+        ],
+        data: [
+          { aa: 'aa0', ab: 'ab0', ac: 'ac0', ad: 'ad0' },
+          { aa: 'aa1', ab: 'ab1', ac: 'ac1', ad: 'ad1' },
+          { aa: 'aa2', ab: 'ab2', ac: 'ac2', ad: 'ad2' },
+          { aa: 'aa3', ab: 'ab3', ac: 'ac3', ad: 'ad3' }
+        ],
+        defaultSort: { key: 'aa', descending: true },
+        sortable: true
+      };
+
+      const app = mountApp(props);
+
+      app.setState({
+        data: [...props.data, { aa: 'aa4', ab: 'ab4', ac: 'ac4', ad: 'ad4' }]
+      });
+
+      const sortedData = app.find(TableBase).props().data;
+
+      expect(sortedData).toMatchSnapshot();
+    });
+  });
```

Comment:
These tests cover a lot of different functionality in a clear & concise way. I appreciate this work.